### ISSUE TITLE
CC0116: Left and right side of assignment must not be the same entity

### ISSUE THREAD

**Issue by kwhitefoot** [code-cracker/code-cracker#746]
#New analyzer:

In VB.Net if you follow the MS recommended naming convention you will almost certainly end up with classes that look a bit like this:

Before:
````vb.net
Class A

Public B as double

public Sub New(b as double)
  B = b
end Sub
````

After:

````vb.net
Class A

Public B as double

public Sub New(b as double)
  Me.B = b
end Sub
````

The problem is that the compiler will not complain.

Diagnostic Id: `CC0116` 
Category: `Naming` 
Severity: `Error`

**Comment by jwooley** [comment]
Good idea. Essentially analyzer should catch if constructor passes in parameter with same name (potentially with different casing as module level fields and recommend disambiguating with Me. Is this something you want to try?

**Comment by kwhitefoot** [comment]
That's right.

Although I used a constructor as the example it applies also to methods.  Of course if the method is in a module or is declared Shared then the fix does not apply so perhaps there should be a separate analyser for that case.

**Comment by giggio** [comment]
OMG, the C# compiler catches this. Nice analyzer!
